DataManagementSystem/scripts/dirac-dms-directory-sync.py

```diff
@@ -0,0 +1,552 @@
+
+  for mydirectory in directories:
+    return_directories.add(mydirectory.replace(path,'').lstrip('/'))
+
+  tree = {}
+  tree["Directories"]=return_directories
+  tree["Files"]=return_files
+
+  return S_OK(tree)
+
+def isInFileCatalog(fc, path ):
+  """
+  Check if the file is in the File Catalog
+  """
+
+  result = fc.listDirectory(path)
+  if result['OK']:
+    if result['Value']['Successful']:
+      return S_OK()
+    else:
+      return S_ERROR()
+  else:
+    return S_ERROR()
+
+def getContentToSync(upload, fc, source_dir, dest_dir):
+  """
+  Return list of files and directories to be create and deleted
+  """
+
+  if upload:
+    res = getSetOfRemoteDirectoriesAndFiles(fc, dest_dir)
+    if not res['OK']:
+      return S_ERROR(res['Message'])
+    to_dirs = res['Value']['Directories']
+    to_files = res['Value']['Files']
+
+    res = getSetOfLocalDirectoriesAndFiles(source_dir)
+    if not res['OK']:
+      return S_ERROR(res['Message'])
+    from_dirs = res['Value']['Directories']
+    from_files = res['Value']['Files']
+
+  else:
+    res = getSetOfLocalDirectoriesAndFiles(dest_dir)
+    if not res['OK']:
+      return S_ERROR(res['Message'])
+    to_dirs = res['Value']['Directories']
+    to_files = res['Value']['Files']
+
+    res = getSetOfRemoteDirectoriesAndFiles(fc, source_dir)
+    if not res['OK']:
+      return S_ERROR(res['Message'])
+    from_dirs = res['Value']['Directories']
+    from_files = res['Value']['Files']
+
+  #Create list of directories to delete
+  dirs_delete = list(to_dirs - from_dirs)
+  #Sort the list by depth of directory tree
+  dirs_delete.sort(key = lambda s: -s.count('/'))
+  #Create list of directories to create
+  dirs_create = list(from_dirs - to_dirs)
+  #Sort the list by depth of directory tree
+  dirs_create.sort(key = lambda s: s.count('/'))
+
+  #Flatten the list of pairs (filename, size) to list of filename
+  files_delete = [pair[0] for pair in list(to_files - from_files)]
+  files_create = [pair[0] for pair in list(from_files - to_files)]
+
+  create = {}
+  create["Directories"] = dirs_create
+  create["Files"] = files_create
+
+  delete = {}
+  delete["Directories"] = dirs_delete
+  delete["Files"] = files_delete
+
+  tree = {}
+  tree["Create"]=create
+  tree["Delete"]=delete
+
+  return S_OK(tree)
+
+
+def removeRemoteFiles(dm,lfns):
+  """
+  Remove file from the catalog
+  """
+  for lfnList in breakListIntoChunks( lfns, 100 ):
+    res = dm.removeFile( lfnList )
+    if not res['OK']:
+      return S_ERROR( "Failed to remove files:" + lfnList + res['Message'] )
+    else:
+      return S_OK()
+
+
+def uploadLocalFile(dm, lfn, localfile, storage):
+  """
+  Upload a local file to a storage element
+  """
+  res = dm.putAndRegister( lfn, localfile, storage, None )
+  if not res['OK']:
+    return S_ERROR( 'Error: failed to upload %s to %s' % ( lfn, storage ) )
+  else:
+    return S_OK( 'Successfully uploaded file to %s' % storage )
+
+def downloadRemoteFile(dm, lfn, destination):
```

Comment:
Why do you need a separate method for downloading/uploading which is exactly the same as the DataManager methods ?